sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,296 @@
+
+ * `accept_downstream_stall`: Provides the compiler the guarantee that
+    downstream logic will back-pressure the kernel.
+
+ * `remove_downstream_stall`: Provides the compiler the guarantee that
+    downstream logic will not back-pressure the kernel.
+
+ If the `streaming_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `streaming_interface_accept_downstream_stall`,
+ `streaming_interface_remove_downstream_stall`.
+
+|`register_map_interface`
+|The `register_map_interface` property adds the requirement that the kernel must have
+ its input / output control and kernel argument data signals placed in a shared
+ Control and Status Register (CSR) map. The following values are supported:
+
+ * `wait_for_done_read`: Provides the compiler the guarantee that downstream
+    logic will back-pressure the kernel.
+
+ * `do_not_wait_for_done_read`: Provides the compiler the guarantee that
+    downstream logic will not back-pressure the kernel.
+ 
+ If the `register_map_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `register_map_interface_wait_for_done_read`,
+ `register_map_interface_do_not_wait_for_done_read`.
+
+|`pipelined`
+|Two sets of values are accepted as property parameters - a boolean, and another
+ property `initiation_interval`.
+
+ When the boolean parameter is set to `true`, the property directs the compiler
+ to pipeline calls to the kernel such that multiple invocations of the kernel
+ can be in flight simultaneously. When the boolean paramter is set to `false`,
+ the compiler will not pipeline kernel invocations.
+
+ The second property argument should be used to specify how many cycles can
+ exist between multiple invocations of the kernel. The `initiation_interval`
+ property should be used to specify this:
+
+ * `true, intiation_interval<N>`: The compiler will pipeline multiple kernel
+   invocations such that an invocation is launched every N cycles.
+ 
+ If the `initiation_interval` property argument is not specified, the compiler
+ will attempt to launch the kernel invocations as fast as it can.
+ 
+ If the `pipelined` property is not specified, the default behavior is
+ equivalent to a combination of the property parameter values listed above, but
+ the choice is implementation defined.
+|===
+
+Device compilers that do not support this extension may accept and ignore these
+ properties.
+
+== Adding a Property List to a Kernel Launch
+
+A simple example of using this extension to launch a kernel with a streaming
+ interface is shown below.
+
+The example assumes that the kernel will not be back-pressured by the downstream
+logic and hence uses the property `stall_free_return_kernel`:
```

Comment:
This is the old name and also the example below does accept a stall so these need to be made consistent.